app/user_views.py

```diff
@@ -1,42 +1,60 @@
-from flask_login import logout_user, current_user, login_required
-from flask import render_template, url_for, redirect, flash, request
+from flask_login import current_user, login_required
+from flask import render_template, url_for, redirect, flash, request, abort
 from flask import current_app as app
 from sqlalchemy import insert
-from .models import db, Rehearsal
-from .forms import RehearsalForm
+from .models import db, Rehearsal, User
+from .forms import RehearsalForm, OrderForm
 import datetime as dt
 
 
-@app.route('/logout')
-def logout():
-    logout_user()
-    return redirect(url_for('index'))
-
-
-@app.route("/all-rehearsals")
+@app.route("/all-rehearsals", methods=["GET", "POST"])
 @login_required
 def get_all_rehearsals():
-    rehearsals = Rehearsal.query.all()
+    # Check date
+    date_check = dt.datetime.today().date()
+    today = date_check.strftime("%a %d %b, %Y")
+
+    form = OrderForm()
+    order_by = request.args.get("order_by")
+    if not order_by:
+        # Handle the case when the user does not select any option
+        # redirect the user to a default ordering
+        return redirect(url_for("get_all_rehearsals", order_by="desc"))
+    elif order_by not in ["asc", "desc", "created"]:
+        # Handle the case when the user is trying to manipulate the order_by parameter
+        # return an error message
+        flash("Invalid value for the order_by parameter")
+        return redirect(url_for("get_all_rehearsals", order_by="desc"))
+    else:
+
+        if order_by == "desc":
+            order_by_clause = Rehearsal.date.desc()
+        elif order_by == "created":
+            order_by_clause = Rehearsal.user_id.asc()
```

Comment:
I'm unsure what `created` means here -- is that like "date created" or "user who created it". You're sorting by `user_id` here. Did you mean to sort by `date`?

Reply:
I was trying to allow the user the ability to sort the rehearsals by the order they created them. Which now I realize is rather unnecessary. So I will remove that 😅